core/src/main/java/hudson/util/SubClassGenerator.java

```diff
@@ -25,17 +25,17 @@
 
 import hudson.PluginManager.UberClassLoader;
 import jenkins.model.Jenkins;
-import org.kohsuke.asm5.ClassWriter;
-import org.kohsuke.asm5.MethodVisitor;
-import org.kohsuke.asm5.Type;
+import org.objectweb.asm.ClassWriter;
+import org.objectweb.asm.MethodVisitor;
+import org.objectweb.asm.Type;
 
 import java.lang.reflect.Constructor;
 
-import static org.kohsuke.asm5.Opcodes.ACC_PUBLIC;
-import static org.kohsuke.asm5.Opcodes.ALOAD;
-import static org.kohsuke.asm5.Opcodes.ILOAD;
-import static org.kohsuke.asm5.Opcodes.INVOKESPECIAL;
-import static org.kohsuke.asm5.Opcodes.RETURN;
+import static org.objectweb.asm.Opcodes.ACC_PUBLIC;
+import static org.objectweb.asm.Opcodes.ALOAD;
+import static org.objectweb.asm.Opcodes.ILOAD;
+import static org.objectweb.asm.Opcodes.INVOKESPECIAL;
+import static org.objectweb.asm.Opcodes.RETURN;
 
 /**
  * Generates a new class that just defines constructors into the super types.
```

Comment:
As in the earlier PR, I think this thing is unused. We should strive to eliminate uses of ASM wherever possible.

Reply:
Agreed that this class should be removed and verified that there are no references with `usage-in-plugins`. Since this is supposed to be the conservative PR I'll mark it with `@Deprecated` in this PR and save the removal for the more aggressive PR (#5524).